Step12-2 passport local strategy를 이용한 로그인 구현

gminiy opened the pull request:
다음은 passport 를 이용하여 jwt, Oauth 구현하겠습니다.

gminiy committed:
project init

gminiy committed:
[feat] server init

- framework : express
- logger : morgan

gminiy committed:
[feat] connect to MongoDB

- db configuaration : ./config/dbConfig.js

gminiy committed:
[feat] <model> user resgist 구현

- User schema
- create function : email, password 를 갖는 user model 생성 후 save db.
- unit test function 작성
  - Model.prototype.save mock function 작성
  - input argument 와 return 이 같은지 확인

gminiy committed:
Merge branch 'step12-2' of https://github.com/gminiy/nodejs-news into step12-2

gminiy committed:
config git commit 삭제

gminiy committed:
[feat] User schema update

- User schema nickname 추가

gminiy committed:
[feat] 회원가입 구현

- URL : POST /auth/register
- routes/auth/index
- auth-contorller
  - regist(req,res)
- unit test
  - Model.save mock function
  - mockRequest, mockResponse
  - 등록 성공시 mockUserinfo와 sentData가 같은지 확인
  - email 없이 시도하면 error 발생하는지 확인

gminiy committed:
[feat] 비밀번호 암호화 저장

gminiy committed:
[refactor]db connection 분리

gminiy committed:
[feat] session store

- mongodb 에 session 저장

gminiy committed:
[feat] passport 를 사용한 로그인 구현

- src/passport.js 에 passport logic
- auth-controller.authenticate return authenticate method로 route 구현
- passport 의존성 주입을 위해 route function 구현 (index)
- login : POST /auth/login

gminiy committed:
[feat] logout 구현

gminiy committed:
Update README.md

gminiy committed:
[feat] test를 위한 static file

gminiy committed:
Merge branch 'step12-2' of https://github.com/gminiy/nodejs-news into step12-2

gminiy committed:
[feat] bootsrap 및 static file 수정

honux77 commented on app.js:
```diff
@@ -0,0 +1,34 @@
+const path = require('path');
+const app = express();
+const session = require('express-session');
+const MongoStore = require('connect-mongo')(session);
```
이거 왜 썼어요??

honux77 commented on routes/auth/auth-controller.js:
```diff
@@ -0,0 +1,42 @@
+const User = require('../../model/user');
+
+module.exports = {
+    regist : (request, response) => {
```
register?

honux77 commented on src/passport.js:
```diff
@@ -0,0 +1,38 @@
+    app.use(passport.initialize());
+    app.use(passport.session());
+
+    passport.serializeUser((user, done) => done(null, user.email));
```
이렇게 할 필요 없어요.

honux77 reviewed (changes requested):
mongostore를 사용하는 상태에서 serial / deserial을 호출하는 코드가 틀렸습니다.
잘 생각해 보고 다시 구현해 보세요.

gminiy committed:
[refactor] passport serial/deserial 변경

- user email 정보로 deserial 시 User find 하던것을 삭제
- user object를 session 에 저장
- regist => register
